monai/inferers/inferer.py

```diff
@@ -64,6 +67,190 @@ def __call__(self, inputs: torch.Tensor, network: Callable, *args: Any, **kwargs
         raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
 
 
+class PatchInferer(Inferer):
+    """
+    Inference on patches instead of the whole image based on Splitter and Merger.
+    This splits the input image into patches and then merge the resulted patches.
+
+    Args:
+        splitter: a `Splitter` object that split the inputs into patches. Defaults to None.
+            If not provided or None, the inputs are considered to be already split into patches.
+        merger: a `Merger` object that merges patch outputs. Defaults to `AvgMerger`.
+        batch_size: batch size for patches. If the input tensor is already batched [BxCxWxH],
+            this adds additional batching [(Bp*B)xCxWpxHp] for inference on patches. Defaults to 1.
+        pre_processor: a callable that process patches before the being fed to the network. Defaults to None.
+        post_processor: a callable that process the output of the network. Defaults to None.
+        output_keys: if the network output is a dictionary, this defines the keys of the output dictionary to use.
+            Defaults to None, where all the keys are taken if output is a dictionary.
+    """
+
+    def __init__(
+        self,
+        splitter: Splitter | None = None,
+        merger: Merger | Sequence[Merger] | None = None,
+        batch_size: int = 1,
+        pre_processor: Callable | None = None,
+        post_processor: Callable | None = None,
+        output_keys: Sequence | None = None,
+    ) -> None:
+        Inferer.__init__(self)
+
+        # splitter
+        if splitter is not None and not callable(splitter):
+            raise TypeError(f"'splitter' should be a callable object, {type(splitter)} is given.")
+        self.splitter = splitter
+
+        # mergers
+        self.mergers: Sequence[Merger] = (AvgMerger(),) if merger is None else ensure_tuple(merger)
+        for m in self.mergers:
+            if not isinstance(m, Merger):
+                raise TypeError(f"'merger' should be a `Merger` object, {type(m)} is given.")
```

Comment:
```suggestion
        if any(not isinstance(m, Merger) for m in self.mergers):
            raise TypeError(f"`merger` should be a `Merger` object or sequence thereof, object of type {type(m)} given.")
```

Reply:
Since we want to include `type(m)` in the error message, it might be better to keep it as it to avoid going over mergers to find the erroneous one.

Reply:
I changed the input in the last update that makes this irrelevant.